File: NativeAPI/Python/b2_create_key.py
import requests
import json

api_url = ""  # Provided by b2_authorize_account
account_id = ""  # Provided by b2_authorize_account
account_authorization_token = ""  # Provided by b2_authorize_account
capabilities = []  # Array of key capabilities
key_name = ""  # Name for the key
valid_duration_in_seconds = 0  # Validity period, positive integer less than 86,400,000 (optional)
bucket_ids = []  # Restrict key to these buckets (optional)
name_prefix = ""  # Restrict access to files with names starting with this prefix (optional)

response = requests.post(
    f"{api_url}/b2api/v4/b2_create_key",
    headers={"Authorization": account_authorization_token},
    json={
        "accountId": account_id,
        "capabilities": capabilities,
        "keyName": key_name,
        "validDurationInSeconds": valid_duration_in_seconds,
        "bucketIds": bucket_ids,
        "namePrefix": name_prefix
    }
)
response_data = response.json()
print(json.dumps(response_data, indent=2))

File: NativeAPI/Java/B2CreateKey.java
package com.backblaze.examples;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class B2CreateKey {
    public static void main(String[] args) {
        String apiUrl = ""; // Provided by b2_authorize_account
        String accountAuthorizationToken = ""; // Provided by b2_authorize_account
        String accountId = ""; // Provided by b2_authorize_account
        String capabilities = ""; // JSON array of key capabilities
        String keyName = ""; // Name for the key
        int validDurationInSeconds = 0; // Validity period, positive integer less than 86,400,000 (optional)
        String bucketId = ""; // Restrict key to this bucket (optional)
        String namePrefix = ""; // Restrict access to files with names starting with this prefix (optional)

        String postParams = "{" +
                "\"accountId\": \"" + accountId + "\"," +
                "\"capabilities\": " + capabilities + "," +
                "\"keyName\": \"" + keyName + "\"," +
                "\"validDurationInSeconds\": " + validDurationInSeconds + "," +
                "\"bucketIds\": [\"" + bucketId + "\"]," +
                "\"namePrefix\": \"" + namePrefix + "\"" +
                "}";

        try (HttpClient client = HttpClient.newHttpClient()) {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(apiUrl + "/b2api/v4/b2_create_key"))
                    .header("Authorization", accountAuthorizationToken)
                    .POST(HttpRequest.BodyPublishers.ofString(postParams))
                    .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            String jsonResponse = response.body();
            System.out.println(jsonResponse);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
